dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/subsystem/SubsystemFeatureDefinitionImpl.java

```diff
@@ -484,6 +484,33 @@ public String toString() {
         return icons;
     }
 
+    /*
+     * (non-Javadoc)
+     *
+     * @see com.ibm.ws.kernel.feature.provisioning.ProvisioningFeatureDefinition#getEndpointIcons()
+     */
+    @Override
+    public Collection<String> getEndpointIcons() {
+        Collection<String> icons = new ArrayList<String>();
+        String endpointIconsHeader = getHeader("Subsystem-Endpoint-Icons");
+        if (endpointIconsHeader != null) {
+            String[] endpointIcons = endpointIconsHeader.split(",");
+            String endpointIconCategory = "";
+            for (String endpointIcon : endpointIcons) {
+                String[] endpointIconDirectives = endpointIcon.split(";");
+                if (endpointIconDirectives.length <= 1 && endpointIcon.indexOf("=") >= 0) {
```

Comment:
Would it be better to assume any first element of a semi-colon `;` separated list which contains an `=` specifies the category?

Reply:
Sure.  I had considered doing the same earlier, and using an if-else to better indicate which part of the icon metadata is processed.  Also, I figured out that "category" is really the "endpoint-name".  You'll that change as well.

Reply:
Changes completed.